docs/reference/modules/node.asciidoc

```diff
@@ -1,32 +1,48 @@
 [[modules-node]]
 == Node
 
-*elasticsearch* allows to configure a node to either be allowed to store
-data locally or not. Storing data locally basically means that shards of
-different indices are allowed to be allocated on that node. By default,
-each node is considered to be a data node, and it can be turned off by
-setting `node.data` to `false`.
+*elasticsearch* allows the configuration of a node to play multiple
+roles. By default, a node's role is a _data node_, a _master node_
+and a _client node_.
 
-This is a powerful setting allowing to simply create smart load
-balancers that take part in some of different API processing. Lets take
-an example:
+*data nodes* store indexed data locally and make it available for
+further manipulation. Storing data locally basically means that
+shards of different indices are allowed to be allocated on that
+node. The default is `node.data: true`, but can also be set to
+`node.data: false` in the configuration file.
 
-We can start a whole cluster of data nodes which do not even start an
-HTTP transport by setting `http.enabled` to `false`. Such nodes will
-communicate with one another using the
-<<modules-transport,transport>> module. In front
-of the cluster we can start one or more "non data" nodes which will
-start with HTTP enabled. All HTTP communication will be performed
-through these "non data" nodes.
+*master nodes* control cluster state, including changes such as node
+addition or removal, index creation and deletion and mapping updates.
+While there may be multiple nodes with this setting enabled, in
+any cluster there can be only one _active master node_ at a time and
+if the current master becomes unavailable then the other master nodes
+can re-elect a new master, ensuring cluster continuity. If you have
+dedicated master nodes then you should not send index or query
+requests to them, this ensures they can maintain the cluster state
+with little to no interruption, ensuring the cluster remains as
+stable as possible. The default is `node.master: true` and
+can also be set to `node.master: false` in the configuration file.
 
-The benefit of using that is first the ability to create smart load
-balancers. These "non data" nodes are still part of the cluster, and
-they redirect operations exactly to the node that holds the relevant
-data. The other benefit is the fact that for scatter / gather based
-operations (such as search), these nodes will take part of the
-processing since they will start the scatter process, and perform the
-actual gather processing.
+*client nodes* create smart load balancers that take part in some of
```

Comment:
*client nodes* are smart load balancers that keep a copy of the cluster state and use it to route request exactly where they need to go. They also act the *coordinating node* for requests sent to them, forwarding the requests to the data nodes and then merging the results when they come back. Remember that by default all nodes act as client nodes so all nodes can be the coordinating node for queries.

I don't understand the bit you have about `node.client:true` implying `node.master: false` and `node.data: false`. Especially in light of every node being a client node by default. Is this just some funky quirk of that setting?